### PR TITLE
add death on exit course define

### PR DESCRIPTION
**Pull request by Reonu**
if the define is enabled, exiting course is the same as dying (warps to failure warp node)

### CODE REVIEW

File: src/game/level_update.c
Comment (gheskett) [inline src/game/level_update.c:1036]: I wouldn't indent preprocessor checks, for the sake of matching the format of the rest of the repository. The main exception to this IMO is in safeguards because they're littered everywhere in that file.

File: src/game/level_update.c
Comment (gheskett) [inline src/game/level_update.c:1038]: I don't think you want this line. In general, what I would suggest doing for exit course death is copy the unpausing code exactly as-is, and follow it up with the `level_trigger_warp` call. This way it should function pretty much exactly as a standard death.

**Review by gheskett — CHANGES_REQUESTED** [review]

**Comment by gheskett** [comment]
I might also suggest adding a redefinition of the exit course text, though managing that around ASCII/multilang might be annoying.

**Comment by thecozies** [comment]
> I might also suggest adding a redefinition of the exit course text, though managing that around ASCII/multilang might be annoying.

I thought the point of this was to make sure that exiting course refills your health but also let's you exit the same way you would for a death. you're still exiting course, might just need to add a life (or prevent losing one) if the dev has lives enabled

**Comment by gheskett** [comment]
Well if it's supposed to act as a death warp, that doesn't necessarily guarantee the destination is outside of said level or even area. But at that point if it's functionally the same as dying, then it would be more accurate to say something along those lines than strictly exiting course.

**Comment by thecozies** [comment]
imo thats up to the creator on what they choose to do, most of the time itll exit level so it works keeping it as-is

**Comment by gheskett** [comment]
I'll leave that up to Reonu I guess

**Review by gheskett — APPROVED** [review]

**Review by thecozies — APPROVED** [review]
I approve as long as we don't squash

**Comment by gheskett** [comment]
> I approve as long as we don't squash

peterVOID

**Pull request merged**
